File: solutions/0771. Jewels and Stones/0771.java
class Solution {
  public int numJewelsInStones(String J, String S) {
    int ans = 0;
    Set<Character> jewels = new HashSet<>();

    for (char j : J.toCharArray())
      jewels.add(j);

    for (final char s : S.toCharArray())
      if (jewels.contains(s))
        ++ans;

    return ans;
  }
}

File: solutions/0771. Jewels and Stones/0771.cpp
class Solution {
 public:
  int numJewelsInStones(string jewels, string stones) {
    int ans = 0;
    unordered_set<char> jewelsSet(jewels.begin(), jewels.end());

    for (const char stone : stones)
      if (jewelsSet.count(stone))
        ++ans;

    return ans;
  }
};
